doc/decisions/constructor_functions.md

```diff
@@ -0,0 +1,85 @@
+# Constructor Functions
+
+## Problem
+
+The constructor functions that `libelektra-core` defines for creating Keys and KeySets must be possible to implement in other languages without complicated workarounds.
```

Comment:
Afaik this problem does not exist, it was possible for every language we support?

```suggestion
The constructor functions for creating Keys and KeySets must be easy to be invoked from other languages without complicated workarounds.
```